### DISCUSSION TITLE
PyTorchClassifier for a pre-trained model (trained with normal pytorch) does not work properly (low accuracy)

### DISCUSSION THREAD

**Discussion by vietvo89** [Trusted-AI/adversarial-robustness-toolbox#1972] [Q&A]
**Describe the bug**
PyTorchClassifier for pre-trained model does not work properly. I tested it on CIFAR10 test set and the accuracy is less than 50%. What is wrong with  PyTorchClassifier and my pre-trained model?

**To Reproduce**
```python
def load_model(net):
    model_path = './models/cifar10_ResNet18.pth'
    net = ResNet18()
    device = 'cuda:0' if torch.cuda.is_available() else 'cpu'
    net = net.to(device)
    checkpoint = torch.load(model_path,map_location='cuda:0')
    if 'net' in checkpoint:
        if device == 'cuda:0':
            net = torch.nn.DataParallel(net)
            cudnn.benchmark = True
        net.load_state_dict(checkpoint['net'])
    else:
        net.load_state_dict(checkpoint)
    net.eval()
    return net

def test_model(model):
    correct = 0
    total = 0
    with torch.no_grad():
        for idx, inputs in tqdm(enumerate(x_test)):
            targets = np.where(y_test[idx]==1)[0]
            outputs = model.predict(np.array([inputs]))
            predicted = np.argmax(outputs,axis=1)
            total += 1
            correct += 1 if predicted==targets else 0   
            if idx%1000==0 and idx>0:
                print(idx,':',correct/total*100.0)
    acc = 100.0 * correct / total
# --------------------------------------------------------------------------
#1. Load model
net = load_model()

#2. Load dataset
(x_train, y_train), (x_test, y_test), min_pixel_value, max_pixel_value = load_cifar10()

#3. Draft my model using PyTorchClassifier
loss_fn = torch.nn.CrossEntropyLoss()
optimizer = torch.optim.Adam(net.parameters(), lr=0.01)

classifier = PyTorchClassifier(
        model=net,
        loss=loss_fn,
        optimizer=optimizer,
        input_shape=(3,32,32),
        nb_classes=10,
        clip_values=(min_pixel_value, max_pixel_value), #clip_values=(0, 255),
        channels_first=True,
    )

# 4. Test model
test_model(classifier)

```
**Expected behavior**
My pre-trained model is able to achieve 95% accuracy on Testset of CIFAR-10. I expect PytorchClassifier with my pre-trained model should achieve the same acc.

**Screenshots**

**System information (please complete the following information):**
 - OS: Ubuntu 
 - Python version: 3.8
 - ART version or commit number: 1.12.1
 - PyTorch version: 1.11
